### PR TITLE
Resolves 2333 "Specify Puppet version requirement"

### PR DESCRIPTION
**Pull request by maximiliankolb**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

**Comment by maximiliankolb** [comment]
After review and merging, we think which commits to cherry-pick, and which to cherry-pick and adapt to ensure that "Foreman version <-> supported Puppet version(s)" matches.

File: guides/common/modules/ref_supported-puppet-versions-and-system-requirements.adoc [hunk 1]
```diff
@@ -4,9 +4,8 @@
 Before you begin with the Puppet integration, review the supported Puppet versions and system requirements.
 
 Supported Puppet Versions::
-{Project} supports the following Puppet versions:
-
-* Puppet 7
+{Project} supports Puppet 7.
+Ensure that your Puppet modules to configure managed hosts are compatible to Puppet 7.
```
Comment (mjivraja) [inline guides/common/modules/ref_supported-puppet-versions-and-system-requirements.adoc:8]:
```suggestion
Ensure that the Puppet modules you use to configure managed hosts are compatible with Puppet 7.
```
Even simplify.

File: guides/doc-Quickstart/master.adoc [hunk 2]
```diff
@@ -22,7 +22,7 @@ include::common/modules/ref_supported-operating-systems.adoc[]
 The installation will require 4GB of memory.
 For more information, see {InstallingServerDocURL}system-requirements_{project-context}[System Requirements].
 
-The Foreman installer uses Puppet (5 or later required) to install Foreman. This guide assumes that you have a newly installed operating system, on which the installer will setup Foreman, a Puppet server, and the Smart Proxy by default. It’s not advisable to follow the steps below on an existing system, since the installer will affect the configuration of several components.
+The Foreman installer uses Puppet to install Foreman. This guide assumes that you have a newly installed operating system, on which the installer will setup Foreman, a Puppet server, and the Smart Proxy by default. It’s not advisable to follow the steps below on an existing system, since the installer will affect the configuration of several components.
```
Comment (mjivraja) [inline guides/doc-Quickstart/master.adoc:25]: Can we follow one sentence per line here?

File: guides/common/modules/ref_supported-puppet-versions-and-system-requirements.adoc (hunk 1 above)
Comment (ekohl) [reply guides/common/modules/ref_supported-puppet-versions-and-system-requirements.adoc:8]: Just wondering if this is even better:
```suggestion
Ensure that the Puppet modules used to configure managed hosts are compatible with Puppet 7.
```

Comment (maximiliankolb) [reply guides/common/modules/ref_supported-puppet-versions-and-system-requirements.adoc]: Applied.

File: guides/doc-Quickstart/master.adoc (hunk 2 above)
Comment (maximiliankolb) [reply guides/doc-Quickstart/master.adoc]: Applied in a separate commit.

**Review by maximiliankolb — COMMENTED** [review]
Applied feedback and rebased to master.

File: guides/doc-Quickstart/master.adoc [hunk 3]
```diff
@@ -13,24 +13,30 @@ include::common/modules/snip_guide-not-ready.adoc[]
 endif::[]
 ifndef::HideDocumentOnStable[]
 
-The Foreman installer is a collection of Puppet modules that installs everything required for a full working Foreman setup. It uses native OS packaging (e.g. RPM and .deb packages) and adds necessary configuration for the complete installation.
+The Foreman installer is a collection of Puppet modules that installs everything required for a full working Foreman setup.
+It uses native OS packaging (e.g. RPM and .deb packages) and adds necessary configuration for the complete installation.
```
Comment (mjivraja) [inline guides/doc-Quickstart/master.adoc:17]:
```suggestion
It uses native OS packaging (e.g. RPM and DEB packages) and adds necessary configuration for the complete installation.
```
Or `.rpm and .deb` packages.

**Review by mjivraja — CHANGES_REQUESTED** [review]
One last and Rest looks good.

File: guides/doc-Quickstart/master.adoc (hunk 3 above)
Comment (maximiliankolb) [reply guides/doc-Quickstart/master.adoc]: Applied and slightly reworded.

File: guides/doc-Quickstart/master.adoc [hunk 4]
```diff
@@ -13,24 +13,30 @@ include::common/modules/snip_guide-not-ready.adoc[]
 endif::[]
 ifndef::HideDocumentOnStable[]
 
-The Foreman installer is a collection of Puppet modules that installs everything required for a full working Foreman setup. It uses native OS packaging (e.g. RPM and .deb packages) and adds necessary configuration for the complete installation.
+The Foreman installer is a collection of Puppet modules that installs everything required for a full working Foreman setup.
+It uses native OS packaging (`.rpm` or `.deb` packages)  and adds necessary configuration for the complete installation.
```
Comment (mjivraja) [inline guides/doc-Quickstart/master.adoc:17]:
```suggestion
It uses native OS packaging (`.rpm` or `.deb` packages) and adds necessary configuration for the complete installation.
```

**Review by mjivraja — APPROVED** [review]
LGTM!.

File: guides/doc-Quickstart/master.adoc (hunk 4 above)
Comment (maximiliankolb) [reply guides/doc-Quickstart/master.adoc]: Argh! Thanks & fixed. :upside_down_face:

**Review by ekohl — APPROVED** [review]

**Pull request merged**